pkg/server/plugin/upstreamauthority/awspca/pca.go

```diff
@@ -230,44 +244,54 @@ func (m *PCAPlugin) MintX509CA(request *upstreamauthorityv0.MintX509CARequest, s
 
 	// The last certificate returned from the chain is the root.
 	upstreamRoot := certChain[len(certChain)-1]
-	bundle := x509util.DedupeCertificates([]*x509.Certificate{upstreamRoot}, m.supplementalBundle)
+	bundle := x509util.DedupeCertificates([]*x509.Certificate{upstreamRoot}, config.supplementalBundle)
 
-	derBundle := x509util.RawCertsFromCertificates(bundle)
+	upstreamX509Roots, err := x509certificate.ToPluginProtos(bundle)
+	if err != nil {
+		return status.Errorf(codes.Internal, "unable to form response upstream X.509 roots: %v", err)
+	}
 
 	// All else comprises the chain (including the issued certificate)
-	derChain := [][]byte{cert.Raw}
-	derChain = append(derChain, x509util.RawCertsFromCertificates(certChain[:len(certChain)-1])...)
+	x509CAChain, err := x509certificate.ToPluginProtos(append([]*x509.Certificate{cert}, upstreamRoot))
```

Comment:
This is dropping the intermediates and adding the root, which is incorrect. The X509CA chain should only contain the CA and any intermediates.
```suggestion
	x509CAChain, err := x509certificate.ToPluginProtos(append([]*x509.Certificate{cert}, certChain[:len(certChain)-1]...))
```